src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -294,6 +296,84 @@ private async Task ExecuteResourceCommandAsync(ResourceViewModel resource, Comma
         }
     }
 
+    private static string? GetSourceTooltip(ResourceViewModel resource)
+    {
+        if (resource.IsProject() && resource.TryGetProjectPath(out var projectPath))
+        {
+            return Path.GetFileName(projectPath);
+        }
+
+        if (resource.TryGetExecutablePath(out var executablePath))
+        {
+            resource.TryGetExecutableArguments(out var arguments);
+            var argumentsString = arguments.IsDefaultOrEmpty ? "" : string.Join(" ", arguments);
+
+            return $"{Path.GetFileName(executablePath)} {argumentsString}";
+        }
+
+        if (resource.TryGetContainerImage(out var containerImage))
+        {
+            return containerImage;
+        }
+
+        if (resource.Properties.TryGetValue(KnownProperties.Resource.Source, out var value) && value.HasStringValue)
+        {
+            return value.StringValue;
+        }
+
+        return null;
+    }
+
+    private string GetEndpointsTooltip(ResourceViewModel resource)
+    {
+        var displayedEndpoints = GetDisplayedEndpoints(resource, out var additionalMessage);
+
+        if (additionalMessage is not null)
+        {
+            return additionalMessage;
+        }
+
+        if (displayedEndpoints.Count == 1)
+        {
+            return displayedEndpoints.First().Text;
+        }
+
+        var maxShownEndpoints = 3;
+        var tooltipBuilder = new StringBuilder(string.Join(", ", displayedEndpoints.Take(maxShownEndpoints).Select(endpoint => endpoint.Text)));
+
+        if (displayedEndpoints.Count > maxShownEndpoints)
+        {
+            tooltipBuilder.Append(CultureInfo.CurrentCulture, $" + {displayedEndpoints.Count - maxShownEndpoints}");
+        }
+
+        return tooltipBuilder.ToString();
```

Comment:
Ditto, de-dupe with `EndpointsColumnDisplay`.

Reply:
The content of the tooltip purposely diverges from the display in EndpointsColumnDisplay, I'm not sure this needs to be 'de-duped'. If so, I can file a follow-up